src/components/MessageList/VirtualizedMessageList.js

```diff
@@ -125,27 +152,14 @@ const VirtualizedMessageList = ({
         totalCount={messages.length}
         overscan={overscan}
         followOutput={true}
-        maxHeightCacheSize={2000} // reset the cache once it reaches 2k
-        scrollSeek={scrollSeekPlaceHolder}
-        item={(i) => messageRenderer(messages, i)}
-        emptyComponent={() => <EmptyStateIndicator listType="message" />}
-        header={() =>
-          loadingMore ? (
-            <div className="str-chat__virtual-list__loading">
-              <LoadingIndicator size={20} />
-            </div>
-          ) : (
-            <></>
-          )
-        }
-        footer={() =>
-          TypingIndicator ? <TypingIndicator avatarSize={24} /> : <></>
-        }
+        itemContent={(i) => messageRenderer(messages, i)}
+        components={virtuosoComponents}
+        firstItemIndex={-numItemsPrepended}
```

Comment:
This one's a bit funny, I'm not sure what's the best way to approach this. The virtuoso examples for this all have a large, mostly empty array as the initial state. As more items are backfilled (i.e. our loadMore function), the examples add these prepended items to the empty array, and the `firstItemIndex` is set to the first index in this mostly empty array that has actual messages. See https://virtuoso.dev/prepend-items/

I'd rather not keep a separate copy of our `messages` array with a bunch of empty slots at the start as state here, that feels very wasteful. I can increase the `firstItemIndex` by some arbitrary amount `x`, only to subtract `x` from the index when we use it to access the `messages` array, but I don't know why that's needed aside from the warning in the virtuoso docs.

To clarify, here's what I'm suggesting we could do:
```
const prependOffset = 10000;

// ....


  const messageRenderer = useCallback(
    (messageList, virtuosoIndex) => {
      const streamMessagesIndex = virtuosoIndex + numItemsPrepended - prependOffset;

// ....

      <Virtuoso
        itemContent={(i) => messageRenderer(messages, i)}
        components={virtuosoComponents}
        firstItemIndex={prependOffset - numItemsPrepended}

```
This would keep the index above 0 for the first 10000 messages that are preloaded. But I'm not sure what the difference would be; it seems to work exactly the same way as the current code.

Reply:
Interesting, I wonder why it specifically says to use positive values here, maybe there are some gotchas we're missing.
but having sth like `const [firstItemIndex, setFirstItemIndex] = useState(50000);` and reducing this number everytime we load more should be good for now

Reply:
Sorry for the inconvenience on that. Internal boundaries are affected by a negative firstItemIndex :(. I will give this some thought, but in the meantime, using an arbitrary large number should work fine.

Reply:
btw great job on v1 @petyosi 👍

Reply:
For sure! I'm not against using a positive index, it just wasn't quite clear to me from the docs why it's needed. I'd love to get a deeper understanding of Virtuoso btw, looks like there's some cool stuff going on under the hood but I find it a little difficult to dig through the source code at the moment because I've never done much with rx-like architectures.